language/tools/test-generation/src/bytecode_generator.rs

```diff
@@ -106,20 +107,31 @@ pub struct FunctionGenerationContext {
     pub function_handle_index: FunctionHandleIndex,
     pub starting_call_height: usize,
     pub locals_len: usize,
+    pub bytecode_len: u64,
 }
 
 impl FunctionGenerationContext {
     pub fn new(
         function_handle_index: FunctionHandleIndex,
         starting_call_height: usize,
         locals_len: usize,
+        bytecode_len: u64,
     ) -> Self {
         Self {
             function_handle_index,
             starting_call_height,
             locals_len,
+            bytecode_len,
         }
     }
+
+    pub fn incr_instruction_count(&mut self) -> Option<()> {
+        self.bytecode_len += 1;
+        if self.bytecode_len >= (u16::max_value() - 1) as u64 {
```

Comment:
See my comment about the maximum bytecode length.  We should have a uniform policy throughout the code about maximum length and offset.  The code in the deserializer is pretty clear about the bounds.

Reply:
Agreed, I'll take a look at this, and publish a PR fixing this and the bug in `get_successors` to correct.